CONFIGURATION.md

```diff
@@ -26,7 +26,7 @@ socket.send.buffer.bytes                 |  *  | 0 .. 100000000  |             0
 socket.receive.buffer.bytes              |  *  | 0 .. 100000000  |             0 | Broker socket receive buffer size. System default is used if 0. <br>*Type: integer*
 socket.keepalive.enable                  |  *  | true, false     |         false | Enable TCP keep-alives (SO_KEEPALIVE) on broker sockets <br>*Type: boolean*
 socket.nagle.disable                     |  *  | true, false     |         false | Disable the Nagle algorithm (TCP_NODELAY) on broker sockets. <br>*Type: boolean*
-socket.max.fails                         |  *  | 0 .. 1000000    |             1 | Disconnect from broker when this number of send failures (e.g., timed out requests) is reached. Disable with 0. NOTE: The connection is automatically re-established. <br>*Type: integer*
+socket.max.fails                         |  *  | 0 .. 1000000    |             1 | Disconnect from broker when this number of send failures (e.g., timed out requests) is reached. Disable with 0. WARNING: It is highly recommended to leave this setting at its default value of 1 to avoid the client and broker to become desynchronized in case of request timeouts. NOTE: The connection is automatically re-established. <br>*Type: integer*
```

Comment:
is it worth considering making this a binary property i.e. "disconnect from broker on send failure"?

Reply:
Yeah, that would probably make more sense, but I'd rather see this property removed altogether. I don't really see a use-case where becoming desynchronized with the broker is something you'd want.   This is a legacy thing..